core/src/main/java/com/linecorp/armeria/server/annotation/JacksonRequestConverterFunction.java

```diff
@@ -68,6 +69,13 @@ public Object convertRequest(ServiceRequestContext ctx, AggregatedHttpMessage re
         final MediaType contentType = request.contentType();
         if (contentType != null && (contentType.is(MediaType.JSON) ||
                                     contentType.subtype().endsWith("+json"))) {
+            if (expectedResultType == byte[].class) {
+                return request.content().array();
```

Comment:
You need to make a copy of the array if the length does not match:

```java
final byte[] array = request.content().array();
final int length = request.content().length();
if (array.length == length) {
    return array;
} else {
    return Arrays.copyOf(array, length);
}
```

@hyangtack I'm curious if we need to try to release the `HttpData` here. What do you think?

Reply:
When the content needs to be converted with `RequestConverterFunction`s, the `AnnotatedHttpService` aggregates the request from [here](https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/internal/annotation/AnnotatedHttpService.java#L223) like the below: 
```
        final CompletableFuture<AggregatedHttpMessage> f =
                aggregationRequired(aggregationStrategy, req) ? req.aggregate()
                                                              : CompletableFuture.completedFuture(null);
```
So it seems not necessary to release the `HttpData` because the aggregator already did it.

Reply:
Thanks for comments. (__)
I have two questions.
1. Does it mean I don't need to handle when `expectedResultType` is the `HttpData`?
2. I don't know the reason why should I copy of the array if the length does not match.
More basically, I don't know well why happen the situation that the length does not match.

Reply:
> I don't know the reason why should I copy of the array if the length does not match.
More basically, I don't know well why happen the situation that the length does not match.

You may refer to [here](https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/common/HttpData.java#L234-L247) about why we need to check the length.
Actually, the array always has the same length with the `content-length` because the aggregator aggregates the contents as a single byte array from [here](https://github.com/line/armeria/blob/master/core/src/main/java/com/linecorp/armeria/common/HttpMessageAggregator.java#L139-L146).

But we need to follow the API specifications, so it would be better to check the length. The current implementation of `ByteArrayRequestConverterFunction` also breaks the rule, so please fix it in your PR. :-)

> Does it mean I don't need to handle when expectedResultType is the HttpData?

For `HttpData`, we just need to assign the object reference to the parameter. So I think we don't need to check the length.